src/Utils/AffineTransform.vala

```diff
@@ -59,8 +88,12 @@ public class Akira.Utils.AffineTransform : Object {
             ((Lib.Items.CanvasArtboard) item).label.translate (delta_x, delta_y);
         }
 
-        initial_event_x = event_x;
-        initial_event_y = event_y;
+        if (!snap_x) {
+            initial_event_x = event_x;
+        }
+        if (!snap_y) {
+            initial_event_y = event_y;
+        }
```

Comment:
I didn't try the updated code, so apologies if my question is silly, but wouldn't this prevent the delta from returning the real mouse acceleration from the user?
We kinda want the snap to be a bit magnetic unless the user moves quickly out of range.
Did you put this in place otherwise the snap is locked?